client/indexer-service/src/handler.rs

```diff
@@ -0,0 +1,183 @@
+use diesel_async::AsyncConnection;
+use futures::prelude::*;
+use log::{error, info};
+use std::sync::Arc;
+use thiserror::Error;
+
+use sc_client_api::{BlockBackend, BlockchainEvents};
+use sp_core::H256;
+use sp_runtime::traits::Header;
+
+use shc_actors_framework::actor::{Actor, ActorEventLoop};
+use shc_common::types::{BlockNumber, ParachainClient};
+use shc_indexer_db::{models::ServiceState, DbConnection, DbPool};
+
+pub(crate) const LOG_TARGET: &str = "indexer-service";
+
+// Since the indexed data should be used directly from the database,
+// we don't need to implement commands.
+#[derive(Debug)]
+pub enum IndexerServiceCommand {}
+
+// The IndexerService actor
+pub struct IndexerService {
+    client: Arc<ParachainClient>,
+    db_pool: DbPool,
+}
+
+// Implement the Actor trait for IndexerService
+impl Actor for IndexerService {
+    type Message = IndexerServiceCommand;
+    type EventLoop = IndexerServiceEventLoop;
+    type EventBusProvider = (); // We're not using an event bus for now
+
+    fn handle_message(
```

Comment:
Any reason why you didn't follow the `blockchain-service/src/handler.rs` structre here? Meaning, having the `ActorEventLoop` implementation at the top, with the `run()` function, then followed by the implementations of `handle_message` and `handle_finality_notification`.

Reply:
No particular reason, I can rearrange them the be consistent. I didn't gave much thought about it since it's lacking an `EventBus`, there are no commands/messages yet (probably won't be) and hence not identical as the blockchain service in this regard.